web/api/report_server.thrift

```diff
@@ -355,6 +355,11 @@ struct SourceComponentData {
 }
 typedef list<SourceComponentData> SourceComponentDataList
 
+struct FailedFileInfo {
```

Comment:
What to you mean by generic way?

Reply:
@gyorb I described it that later it will be easier to extend the API and return not only run specific things such as run name but the analyzer name or version info.

Reply:
I understand that but I'm not sure why it is called FailedFileInfo in that case.
No file information will be contained there, the name value will not be the file name but the run name (and further analyzer specific information).
RunData contains the values you mention codechecker version or more information in the AnalyzerStatistics value.
```
struct RunData {
  1: i64                       runId,                // Unique id of the run.
  2: string                    runDate,              // Date of the run last updated.
  3: string                    name,                 // Human-given identifier.
  4: i64                       duration,             // Duration of the run (-1 if not finished).
  5: i64                       resultCount,          // Number of unresolved results (review status is not FALSE_POSITIVE or INTENTIONAL) in the run.
  6: string                    runCmd,               // The used check command. !!!DEPRECATED!!! This field will be empty so use the getCheckCommand API function to get the check command for a run.
  7: map<DetectionStatus, i32> detectionStatusCount, // Number of reports with a particular detection status.
  8: string                    versionTag,           // Version tag of the latest run.
  9: string                    codeCheckerVersion,   // CodeChecker client version of the latest analysis.
  10: AnalyzerStatisticsData   analyzerStatistics,   // Statistics for analyzers. Only number of failed and successfully analyzed
                                                     // files field will be set. To get full analyzer statistics please use the
                                                     // 'getAnalysisStatistics' API function.
  11: optional string          description,          // A custom textual description.
}
typedef list<RunData> RunDataList
```
The RunData is general enough for these additional values if the api is extended, that is why I do not see why it is worth to introduce another type which will contain similar information what is in RunData right now.

Reply:
It is called FailedFileInfo because it contains information of a failed file. If you know a better name we can rename it to something else.

I don't really wan't to return with a type (RunData), which contains 11 fields but we set only 1 field. Someone who writes a new client and sees only the API and tries to call the getFailedFiles function but we say we return with a RunData he may expect that okay, why the run id or the run command is not set? Why do we set only the run name field of this struct? I wan't to avoid it.

What do you think @bruntib?

Reply:
Me too, I think that `FailedFileInfo` can be ambiguous. This conversation proves that even if you intended having `(FailedFile)Info`, it can be interpreted as `Failed(FileInfo)`. What about `AnalysisFailureInfo`?

Reply:
Okay, it is a better name, I changed it.